File: python/factorial.py
def factorial(x):
    result=1
    for i in range(1,x+1):
        result=result*i
    show(result,x)
        
def main():
    number=int(input("enter number: "))
    factorial(number)


    
def show(result,number):
    print(f'''
    ====================================================
    factorial of {number} is {result}.
    ====================================================
    ''')



main()

File: prolog/factorial.pl
% Base case: factorial of 0 is 1
factorial(0, 1).

% Recursive case: factorial of N is N * factorial of (N - 1)
factorial(N, Result) :-
    N > 0,
    N1 is N - 1,
    factorial(N1, SubResult),
    Result is N * SubResult.
